doc/userguide/output/pcap-alert-output.rst

```diff
@@ -0,0 +1,53 @@
+Pcap Alert Output
+=================
+
+Suricata is able to provide a packet capture output upon the generation of an
+alert. This feature makes use of the "tag" keyword in a signature by searching
+for and dumping tagged packets that either have generated an alert or belong to
+the same session of a packet that has generated an alert. This feature must be
+enabled through the suricata.yaml file, but also requires a signature to have
+the "tag:session;" keyword for it to perform the capture. Signatures without
+"tag:session;" will not trigger a capture.
+
+Additionally, there is an option to enabled a session-dump in the suricata.yaml
+file. This causes all available tcp segments in the relevant session at the time
+of alert to be dumped to the capture file. This is recommended, as otherwise
+often the packet that generates an alert may not be captured due to a lag
+between the processing of a packet and the generation of an alert. This
+behaviour was only found in tcp based traffic. As a result, the session-dump
+option only relates to tcp based traffic.
+
+By default, the stream-pcap-log is not enabled.
```

Comment:
Does this have an noticeable impact on performance? If yes, we should mention that like we do with profiling for example.

Reply:
I can do a bit more analysis on the exact performance implications. There would be two levels of cost/performance differences:
1.  The stream-pcap-log function is enabled, but no tcp session dumping: This would just add in the cost associated with the pcap file setups and the file writes, i.e. the pcap dumps to file.
2. The stream-pcap-log function is enabled and tcp session dumping is too: Same costs associated with the above + some additional information to be held on to in the tcp segments so some additional memory allocations (performed with segment pool initialization) for the storage of this data and then also memcpy'ing the data from packet header into this storage as well. There could also be the case where a Realloc is performed if the packet header storage size is too small.

Reply:
Is there a particular test framework you use / would suggest for evaluating the performance? I have some suricata verify cases that I could put together, but that's more so for validation of the output rather than performance evaluation right?

Reply:
I would recommend testing with t-rex which is a solid way of testing Suricata performance.